### PR TITLE
Update com_github_curl: curl-7_81_0 (current: curl-7_80_0)

### LINKED ISSUE #19389: Newer release available `com_github_curl`: curl-7_81_0 (current: curl-7_80_0)
Package Name: com_github_curl
Current Version: curl-7_80_0@2021-11-10 06:42:29
Available Version: curl-7_81_0@2022-01-05 07:48:46
Upstream releases: https://github.com/curl/curl/releases

### PR DESCRIPTION
**Pull request by kfaseela**
Fixes #19389

Signed-off-by: Faseela K <faseela.k@est.tech>

### CODE REVIEW

**Comment by moderation** [comment]
Interesting CI test failure. Wonder if the following from the [changelog](https://curl.se/changes.html#7_81_0) is the issue?

`openssl: define HAVE_OPENSSL_VERSION for OpenSSL 1.1.0+ `

**Comment by htuch** [comment]
I'm a bit confused why this had SSL as a disabled feature in https://github.com/envoyproxy/envoy/blob/main/test/dependencies/curl_test.cc. @lizan @wrowe you folks come up in the review/authorship history, any thoughts?

**Comment by wrowe** [comment]
Whatever was done, Windows test is still passing, while Linux is now failing, so their intended change seems incomplete in any case.

Examining the CMakeFiles.txt delta between the curl releases should be useful.

**Comment by kfaseela** [comment]
@wrowe pointed me to https://github.com/curl/curl/commit/0969805e20ffe3731a9b020a990a907c3eec907f which mentions that all CMAKE_USE_ has become CURL_USE_.

We would need to adapt to this in and around https://github.com/envoyproxy/envoy/blob/main/bazel/foreign_cc/BUILD#L172

**Comment by wrowe** [comment]
I'm also experimenting with the changes required on another platform, but it appears that this release may have been validated on Ubuntu, and broken builds on RHEL. Just a red flag that this will require extra attention, and we may want to look at the scope of the features before and after. We are testing for a specific subset, yet libcurl keeps baking in more and more dependencies we must not trip over. Something using ldd or on windows, depends.exe or objdump.exe might indicate we have accidently grabbed another autodetected component of libcurl, since these all keep being piled in as opt-out.

**Comment by moderation** [comment]
CI passed so maybe @kfaseela 's latest change did the trick. Move out of draft?

**Comment by kfaseela** [comment]
@htuch I think let us wait for @wrowe 's approval as he is more concerned about other aspects as well with the upgrade. I see that he has created an issue to follow this up.

Also, I was wondering do we still need these curl patches even after upgrade? https://github.com/envoyproxy/envoy/blob/main/bazel/repositories.bzl#L795

File: bazel/foreign_cc/BUILD
Comment (wrowe) [inline bazel/foreign_cc/BUILD:172]: Based on the comment above and the fact that we've updated build toolchains to a more modern flavor of cmake, and curl and openssl have made tweaks here, this may no longer be true - we might not have to dodge this anymore. @lizan when you are back could you comment?

Comment (kfaseela) [reply bazel/foreign_cc/BUILD:172]: @wrowe without this tests are not passing. what do you recommend as the way forward?

**Comment by phlax** [comment]
@kfaseela as this one is passing tests would you un-WIP it while we wait for approval

**Comment by kfaseela** [comment]
> @kfaseela as this one is passing tests would you un-WIP it while we wait for approval

Done, but let us wait for the approval from @wrowe as he knows the consequences better :)

**Review by wrowe — APPROVED** [review]
This looks good @kfaseela TYVM!

https://github.com/appsuite/oss-httpd-build/issues/20 goes into what we are seeing with brotli X openssl breakage in the curl autoconf... but apparently not with cmake bazel builds.

**Pull request merged**
